### ISSUE TITLE
[Bug]: (logging) `fatal error: concurrent map writes` (`Error: Plugin did not respond`)

### ISSUE THREAD

**Issue by jeremychauvet** [hashicorp/terraform-provider-aws#29236]
_Previously titled: "[Bug]: The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call."_

_Changing the name because at this stage I believe it is misleading since **any (?)** error may cause this crash._

_See #29153_

### Terraform Core Version

v1.3.7

### AWS Provider Version

v4.53.0

### Affected Resource(s)

All ressources

### Expected Behavior

Create, update or delete resources

### Actual Behavior

Crash randomly

### Relevant Error/Panic Output Snippet

```shell
[...]
module.xxxxx.aws_security_group.lambdas: Still destroying... [id=sg-xxxxx, 10s elapsed]
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
Stack trace from the terraform-provider-aws_v4.53.0_x5 plugin:
fatal error: concurrent map writes
goroutine 1730 [running]:
github.com/hashicorp/terraform-plugin-log/tflog.SetField.func1(...)
	github.com/hashicorp/terraform-plugin-log@v0.7.0/internal/logging/options.go:212
github.com/hashicorp/terraform-plugin-log/tflog.SetField({0xe25bd00, 0xc001d5f890}, {0xce4900e, 0xb}, {0xa619820?, 0xc00089eb90})
	github.com/hashicorp/terraform-plugin-log@v0.7.0/tflog/provider.go:18 +0x111
github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2.setAWSFields({0xe25bd00?, 0xc003bc12f0?}, 0xc005cbaa00)
	github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2@v2.0.0-beta.23/logger.go:40 +0x9c
github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2.logRequest(0xc005cbaa00)
	github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2@v2.0.0-beta.23/logger.go:69 +0x5f
github.com/aws/aws-sdk-go/aws/request.(*HandlerList).Run(0xc005cbac28, 0xc005cbaa00)
	github.com/aws/aws-sdk-go@v1.44.191/aws/request/handlers.go:267 +0x9a
github.com/aws/aws-sdk-go/aws/request.(*Request).sendRequest(0xc005cbaa00)
	github.com/aws/aws-sdk-go@v1.44.191/aws/request/request.go:602 +0xbf
github.com/aws/aws-sdk-go/aws/request.(*Request).Send(0xc005cbaa00)
	github.com/aws/aws-sdk-go@v1.44.191/aws/request/request.go:555 +0x156
github.com/aws/aws-sdk-go/aws/request.(*Pagination).Next(0xc00501bbf0)
	github.com/aws/aws-sdk-go@v1.44.191/aws/request/request_pagination.go:107 +0xb3
github.com/aws/aws-sdk-go/service/ec2.(*EC2).DescribeNetworkInterfacesPagesWithContext(0xc00423cc40, {0xe25bd00?, 0xc003bc12f0}, 0xc00333e050, 0xc00501bd20, {0x0, 0x0, 0x0})
	github.com/aws/aws-sdk-go@v1.44.191/service/ec2/api.go:26518 +0x19d
github.com/hashicorp/terraform-provider-aws/internal/service/ec2.FindNetworkInterfaces({0xe25bd00?, 0xc003bc12f0?}, 0x48?, 0xc00333e050)
	github.com/hashicorp/terraform-provider-aws/internal/service/ec2/find.go:1526 +0x8b
github.com/hashicorp/terraform-provider-aws/internal/service/ec2.FindNetworkInterface({0xe25bd00?, 0xc003bc12f0?}, 0xc00315b780?, 0xc00333e050)
	github.com/hashicorp/terraform-provider-aws/internal/service/ec2/find.go:1506 +0x2c
github.com/hashicorp/terraform-provider-aws/internal/service/ec2.FindNetworkInterfaceByID({0xe25bd00, 0xc003bc12f0}, 0xc002c92e28?, {0xc006346510, 0x15})
	github.com/hashicorp/terraform-provider-aws/internal/service/ec2/find.go:1559 +0x10c
github.com/hashicorp/terraform-provider-aws/internal/service/ec2.StatusNetworkInterfaceStatus.func1()
	github.com/hashicorp/terraform-provider-aws/internal/service/ec2/status.go:1123 +0x31
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext.func1()
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/resource/state.go:110 +0x1ff
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/resource/state.go:83 +0x1d8
goroutine 1 [select, 1 minutes]:
github.com/hashicorp/go-plugin.Serve(0xc004ad04e0)
	github.com/hashicorp/go-plugin@v1.4.8/server.go:471 +0x14f7
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.Serve({0xcf5e0a9, 0x23}, 0xc0044c5760, {0x0, 0x0, 0x0})
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:312 +0xba5
main.main()
	github.com/hashicorp/terraform-provider-aws/main.go:32 +0x18d
goroutine 321 [select, 1 minutes]:
github.com/hashicorp/go-plugin.(*gRPCBrokerServer).Recv(0xc0000c7fa0?)
	github.com/hashicorp/go-plugin@v1.4.8/grpc_broker.go:121 +0x67
github.com/hashicorp/go-plugin.(*GRPCBroker).Run(0xc004b72140)
	github.com/hashicorp/go-plugin@v1.4.8/grpc_broker.go:411 +0x44
created by github.com/hashicorp/go-plugin.(*GRPCServer).Init
	github.com/hashicorp/go-plugin@v1.4.8/grpc_server.go:85 +0x46a
goroutine 322 [IO wait, 1 minutes]:
internal/poll.runtime_pollWait(0x7f1cfababed0, 0x72)
	runtime/netpoll.go:305 +0x89
internal/poll.(*pollDesc).wait(0xc004ad09c0?, 0xc000a3e000?, 0x1)
	internal/poll/fd_poll_runtime.go:84 +0x32
internal/poll.(*pollDesc).waitRead(...)
	internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc004ad09c0, {0xc000a3e000, 0x1000, 0x1000})
	internal/poll/fd_unix.go:167 +0x25a
os.(*File).read(...)
	os/file_posix.go:31
os.(*File).Read(0xc004a3a820, {0xc000a3e000?, 0x400?, 0xa6ff020?})
	os/file.go:118 +0x5e
bufio.(*Reader).Read(0xc0000c0f40, {0xc000a40000, 0x400, 0x0?})
	bufio/bufio.go:237 +0x1bb
github.com/hashicorp/go-plugin.copyChan({0xe284db0, 0xc0045f9170}, 0x0?, {0xe19eae0?, 0xc004a3a820?})
	github.com/hashicorp/go-plugin@v1.4.8/grpc_stdio.go:181 +0x1f6
created by github.com/hashicorp/go-plugin.newGRPCStdioServer
	github.com/hashicorp/go-plugin@v1.4.8/grpc_stdio.go:37 +0xf5
goroutine 323 [IO wait, 1 minutes]:
internal/poll.runtime_pollWait(0x7f1cfababcf0, 0x72)
	runtime/netpoll.go:305 +0x89
internal/poll.(*pollDesc).wait(0xc004ad0a80?, 0xc00134a000?, 0x1)
	internal/poll/fd_poll_runtime.go:84 +0x32
internal/poll.(*pollDesc).waitRead(...)
	internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc004ad0a80, {0xc00134a000, 0x1000, 0x1000})
	internal/poll/fd_unix.go:167 +0x25a
os.(*File).read(...)
	os/file_posix.go:31
os.(*File).Read(0xc004a3a830, {0xc00134a000?, 0x400?, 0xa6ff020?})
	os/file.go:118 +0x5e
bufio.(*Reader).Read(0xc0000c1740, {0xc00134c000, 0x400, 0x0?})
	bufio/bufio.go:237 +0x1bb
github.com/hashicorp/go-plugin.copyChan({0xe284db0, 0xc0045f9170}, 0x0?, {0xe19eae0?, 0xc004a3a830?})
	github.com/hashicorp/go-plugin@v1.4.8/grpc_stdio.go:181 +0x1f6
created by github.com/hashicorp/go-plugin.newGRPCStdioServer
	github.com/hashicorp/go-plugin@v1.4.8/grpc_stdio.go:38 +0x185
goroutine 337 [syscall, 1 minutes]:
os/signal.signal_recv()
	runtime/sigqueue.go:152 +0x2f
os/signal.loop()
	os/signal/signal_unix.go:23 +0x19
created by os/signal.Notify.func1.1
	os/signal/signal.go:151 +0x2a
goroutine 338 [chan receive, 1 minutes]:
github.com/hashicorp/go-plugin.Serve.func3()
	github.com/hashicorp/go-plugin@v1.4.8/server.go:436 +0x75
Error: The terraform-provider-aws_v4.53.0_x5 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
Cleaning up project directory and file based variables
00:01
ERROR: Job failed: exit code 1
```

### Terraform Configuration Files

N/A

### Steps to Reproduce

1. Update a Terraform stack to v4.53.0 of the provider:

```
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "4.53.0"
    }
  }

  required_version = ">= 1.2"
}
```
2. Terraform plan / apply / destroy

### Panic Output

```
Stack trace from the terraform-provider-aws_v4.53.0_x5 plugin:
fatal error: concurrent map writes
goroutine 1730 [running]:
github.com/hashicorp/terraform-plugin-log/tflog.SetField.func1(...)
	github.com/hashicorp/terraform-plugin-log@v0.7.0/internal/logging/options.go:212
github.com/hashicorp/terraform-plugin-log/tflog.SetField({0xe25bd00, 0xc001d5f890}, {0xce4900e, 0xb}, {0xa619820?, 0xc00089eb90})
	github.com/hashicorp/terraform-plugin-log@v0.7.0/tflog/provider.go:18 +0x111
github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2.setAWSFields({0xe25bd00?, 0xc003bc12f0?}, 0xc005cbaa00)
	github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2@v2.0.0-beta.23/logger.go:40 +0x9c
github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2.logRequest(0xc005cbaa00)
	github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2@v2.0.0-beta.23/logger.go:69 +0x5f
github.com/aws/aws-sdk-go/aws/request.(*HandlerList).Run(0xc005cbac28, 0xc005cbaa00)
	github.com/aws/aws-sdk-go@v1.44.191/aws/request/handlers.go:267 +0x9a
github.com/aws/aws-sdk-go/aws/request.(*Request).sendRequest(0xc005cbaa00)
	github.com/aws/aws-sdk-go@v1.44.191/aws/request/request.go:[602](https://gitlab.eulerhermes.com/deployment/datahub/business-events/eventgen-grade-record/-/jobs/12258660#L602) +0xbf
github.com/aws/aws-sdk-go/aws/request.(*Request).Send(0xc005cbaa00)
	github.com/aws/aws-sdk-go@v1.44.191/aws/request/request.go:555 +0x156
github.com/aws/aws-sdk-go/aws/request.(*Pagination).Next(0xc00501bbf0)
	github.com/aws/aws-sdk-go@v1.44.191/aws/request/request_pagination.go:107 +0xb3
github.com/aws/aws-sdk-go/service/ec2.(*EC2).DescribeNetworkInterfacesPagesWithContext(0xc00423cc40, {0xe25bd00?, 0xc003bc12f0}, 0xc00333e050, 0xc00501bd20, {0x0, 0x0, 0x0})
	github.com/aws/aws-sdk-go@v1.44.191/service/ec2/api.go:26518 +0x19d
github.com/hashicorp/terraform-provider-aws/internal/service/ec2.FindNetworkInterfaces({0xe25bd00?, 0xc003bc12f0?}, 0x48?, 0xc00333e050)
	github.com/hashicorp/terraform-provider-aws/internal/service/ec2/find.go:1526 +0x8b
github.com/hashicorp/terraform-provider-aws/internal/service/ec2.FindNetworkInterface({0xe25bd00?, 0xc003bc12f0?}, 0xc00315b780?, 0xc00333e050)
	github.com/hashicorp/terraform-provider-aws/internal/service/ec2/find.go:1506 +0x2c
github.com/hashicorp/terraform-provider-aws/internal/service/ec2.FindNetworkInterfaceByID({0xe25bd00, 0xc003bc12f0}, 0xc002c92e28?, {0xc006346510, 0x15})
	github.com/hashicorp/terraform-provider-aws/internal/service/ec2/find.go:1559 +0x10c
github.com/hashicorp/terraform-provider-aws/internal/service/ec2.StatusNetworkInterfaceStatus.func1()
	github.com/hashicorp/terraform-provider-aws/internal/service/ec2/status.go:1123 +0x31
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext.func1()
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/resource/state.go:110 +0x1ff
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/resource/state.go:83 +0x1d8
goroutine 1 [select, 1 minutes]:
github.com/hashicorp/go-plugin.Serve(0xc004ad04e0)
	github.com/hashicorp/go-plugin@v1.4.8/server.go:471 +0x14f7
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.Serve({0xcf5e0a9, 0x23}, 0xc0044c5760, {0x0, 0x0, 0x0})
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:312 +0xba5
main.main()
	github.com/hashicorp/terraform-provider-aws/main.go:32 +0x18d
goroutine 321 [select, 1 minutes]:
github.com/hashicorp/go-plugin.(*gRPCBrokerServer).Recv(0xc0000c7fa0?)
	github.com/hashicorp/go-plugin@v1.4.8/grpc_broker.go:121 +0x67
github.com/hashicorp/go-plugin.(*GRPCBroker).Run(0xc004b72140)
	github.com/hashicorp/go-plugin@v1.4.8/grpc_broker.go:411 +0x44
created by github.com/hashicorp/go-plugin.(*GRPCServer).Init
	github.com/hashicorp/go-plugin@v1.4.8/grpc_server.go:85 +0x46a
goroutine 322 [IO wait, 1 minutes]:
internal/poll.runtime_pollWait(0x7f1cfababed0, 0x72)
	runtime/netpoll.go:305 +0x89
internal/poll.(*pollDesc).wait(0xc004ad09c0?, 0xc000a3e000?, 0x1)
	internal/poll/fd_poll_runtime.go:84 +0x32
internal/poll.(*pollDesc).waitRead(...)
	internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc004ad09c0, {0xc000a3e000, 0x1000, 0x1000})
	internal/poll/fd_unix.go:167 +0x25a
os.(*File).read(...)
	os/file_posix.go:31
os.(*File).Read(0xc004a3a820, {0xc000a3e000?, 0x400?, 0xa6ff020?})
	os/file.go:118 +0x5e
bufio.(*Reader).Read(0xc0000c0f40, {0xc000a40000, 0x400, 0x0?})
	bufio/bufio.go:237 +0x1bb
github.com/hashicorp/go-plugin.copyChan({0xe284db0, 0xc0045f9170}, 0x0?, {0xe19eae0?, 0xc004a3a820?})
	github.com/hashicorp/go-plugin@v1.4.8/grpc_stdio.go:181 +0x1f6
created by github.com/hashicorp/go-plugin.newGRPCStdioServer
	github.com/hashicorp/go-plugin@v1.4.8/grpc_stdio.go:37 +0xf5
goroutine 323 [IO wait, 1 minutes]:
internal/poll.runtime_pollWait(0x7f1cfababcf0, 0x72)
	runtime/netpoll.go:305 +0x89
internal/poll.(*pollDesc).wait(0xc004ad0a80?, 0xc00134a000?, 0x1)
	internal/poll/fd_poll_runtime.go:84 +0x32
internal/poll.(*pollDesc).waitRead(...)
	internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc004ad0a80, {0xc00134a000, 0x1000, 0x1000})
	internal/poll/fd_unix.go:167 +0x25a
os.(*File).read(...)
	os/file_posix.go:31
os.(*File).Read(0xc004a3a830, {0xc00134a000?, 0x400?, 0xa6ff020?})
	os/file.go:118 +0x5e
bufio.(*Reader).Read(0xc0000c1740, {0xc00134c000, 0x400, 0x0?})
	bufio/bufio.go:237 +0x1bb
github.com/hashicorp/go-plugin.copyChan({0xe284db0, 0xc0045f9170}, 0x0?, {0xe19eae0?, 0xc004a3a830?})
	github.com/hashicorp/go-plugin@v1.4.8/grpc_stdio.go:181 +0x1f6
created by github.com/hashicorp/go-plugin.newGRPCStdioServer
	github.com/hashicorp/go-plugin@v1.4.8/grpc_stdio.go:38 +0x185
goroutine 337 [syscall, 1 minutes]:
os/signal.signal_recv()
	runtime/sigqueue.go:152 +0x2f
os/signal.loop()
	os/signal/signal_unix.go:23 +0x19
created by os/signal.Notify.func1.1
	os/signal/signal.go:151 +0x2a
goroutine 338 [chan receive, 1 minutes]:
github.com/hashicorp/go-plugin.Serve.func3()
	github.com/hashicorp/go-plugin@v1.4.8/server.go:436 +0x75
Error: The terraform-provider-aws_v4.53.0_x5 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Important Factoids

Reverting in v4.52.0 is a workaround ✅ 

### References

* #29223 
* #29153

### Would you like to implement a fix?

None

**Comment by ewbankkit** [comment]
Similar: https://github.com/hashicorp/terraform-provider-aws/issues/29153.

**Comment by ewbankkit** [comment]
`terraform-plugin-log` issue: https://github.com/hashicorp/terraform-plugin-log/issues/126.

**Comment by YakDriver** [comment]
In case it is helpful, here is a complete trace:

https://gist.github.com/YakDriver/66b508dab3342da50b2da75f4b17a73a

**Comment by ascopes** [comment]
Based on @bflad's comment at https://github.com/hashicorp/terraform-plugin-log/issues/126#issuecomment-1422622939, assuming this is indeed related to that fix, in which case bumping that version will assumably fix this?

**Comment by chrisba11** [comment]
I am also running into this. It's happening primarily when we are running a destroy workflow to clean up feature stack resources. The error occurs while waiting for our RDS cluster to be destroyed. If I re-run the workflow job again right away, it continues to error. Since the RDS database takes about 10-15 minutes to destroy, I've found that if I come back and re-run the workflow after I know the database is destroyed, the error goes away.

**Comment by Agony1023** [comment]
Just ran into the same issue today while doing a `tf destroy`. 
`Error: The terraform-provider-aws_v4.53.0_x5 plugin crashed!`

Added a [terraform.tf](https://developer.hashicorp.com/terraform/tutorials/configuration-language/provider-versioning#explore-terraform-tf) to the project:
```terraform
    aws = {
      source  = "hashicorp/aws"
      version = ">= 4.52.0, != 4.53.0"
    }
```
Skipping 4.53.0 for now.

**Comment by YakDriver** [comment]
This required an upstream fix (hashicorp/terraform-plugin-log#132) which will be part of v4.54.0. We are very sorry for the inconvenience! (Which also greatly impacted us internally 😞 .) Fixed with merged new version of `terraform-plugin-log` in #29269.
